server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -306,61 +287,52 @@ private void registerRequestHandler(DynamicActionRegistry dynamicActionRegistry)
-                logger.info("Loaded all extensions");
-            }
-        } else {
-            logger.warn("Extensions.yml file is not present.  No extensions will be loaded.");
-        }
-    }
-
     /**
      * Loads a single extension
      * @param extension The extension to be loaded
      */
-    private void loadExtension(Extension extension) throws IOException {
-        if (extensionIdMap.containsKey(extension.getUniqueId())) {
-            logger.info("Duplicate uniqueId " + extension.getUniqueId() + ". Did not load extension: " + extension);
-        } else {
-            try {
-                DiscoveryExtensionNode discoveryExtensionNode = new DiscoveryExtensionNode(
-                    extension.getName(),
-                    extension.getUniqueId(),
-                    new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
-                    new HashMap<String, String>(),
-                    Version.fromString(extension.getOpensearchVersion()),
-                    Version.fromString(extension.getMinimumCompatibleVersion()),
-                    extension.getDependencies()
-                );
+    public void loadExtension(Extension extension) throws IOException {
+        try {
+            validateExtension(extension);
+            DiscoveryExtensionNode discoveryExtensionNode = new DiscoveryExtensionNode(
+                extension.getName(),
+                extension.getUniqueId(),
+                new TransportAddress(InetAddress.getByName(extension.getHostAddress()), Integer.parseInt(extension.getPort())),
+                new HashMap<String, String>(),
+                Version.fromString(extension.getOpensearchVersion()),
+                Version.fromString(extension.getMinimumCompatibleVersion()),
+                extension.getDependencies()
+            );
+            extensionIdMap.put(extension.getUniqueId(), discoveryExtensionNode);
+            extensionSettingsMap.put(extension.getUniqueId(), extension);
+            logger.info("Loaded extension with uniqueId " + extension.getUniqueId() + ": " + extension);
+        } catch (IOException e) {
+            throw e;
+        } catch (IllegalArgumentException e) {
+            throw e;
+        }
 
-                extensionIdMap.put(extension.getUniqueId(), discoveryExtensionNode);
-                extensionSettingsMap.put(extension.getUniqueId(), extension);
-                logger.info("Loaded extension with uniqueId " + extension.getUniqueId() + ": " + extension);
-            } catch (OpenSearchException e) {
-                logger.error("Could not load extension with uniqueId " + extension.getUniqueId() + " due to " + e);
-            } catch (IllegalArgumentException e) {
-                throw e;
-            }
+    }
+
+    private boolean validateExtension(Extension extension) throws IOException {
+        if (Strings.isNullOrEmpty(extension.getName())) {
```

Comment:
I really don't like the repetition here.  Can we do something like
```java
void validateField(String fieldName, String value) {
    if (Strings.isNullOrEmpty(value)) {
        throw new IOException("Required field [" + fieldName + "] is missing in the request");
    }
}
```
and then make this method
```java
validateField("name", extension.getName());
validateField("uniqueId", extension.getUniqueId());
validateField("extension host address", extension.getHostAddress());
// ... and so on
```